common/scala/src/main/scala/whisk/core/database/ElasticSearchActivationRestClient.scala

```diff
@@ -0,0 +1,292 @@
+                             kind: Option[String] = None,
+                             cause: Option[String] = None,
+                             causedBy: Option[String] = None,
+                             limits: Option[ActionLimits] = None,
+                             path: Option[String] = None,
+                             logs: ActivationLogs,
+                             waitTime: Option[Int] = None,
+                             initTime: Option[Int] = None) {
+
+    def toActivation() = {
+      val response = statusCode match {
+        case 0 => ActivationResponse.success(Some(result.parseJson.asJsObject))
+        case 1 => ActivationResponse.applicationError(result.parseJson.asJsObject.fields("error"))
+        case 2 => ActivationResponse.containerError(result.parseJson.asJsObject.fields("error"))
+        case 3 => ActivationResponse.whiskError(result.parseJson.asJsObject.fields("error"))
+      }
+      val causedByAnnotation = causedBy.map(value => Parameters("causedBy", value.toJson)).getOrElse(Parameters())
+      val memoryAnnotation = limits
+        .map { value =>
+          Parameters(
+            "limits",
+            JsObject(
+              "memory" -> value.memory.megabytes.toJson,
+              "timeout" -> value.timeout.toJson,
+              "logs" -> value.logs.toJson))
+        }
+        .getOrElse(Parameters())
+      val kindAnnotation = kind.map(value => Parameters("kind", value.toJson)).getOrElse(Parameters())
+      val pathAnnotation = path.map(value => Parameters("path", value.toJson)).getOrElse(Parameters())
+      val waitTimeAnnotation = waitTime.map(value => Parameters("waitTime", value.toJson)).getOrElse(Parameters())
+      val initTimeAnnotation = initTime.map(value => Parameters("initTime", value.toJson)).getOrElse(Parameters())
+
+      WhiskActivation(
+        EntityPath(namespace),
+        EntityName(name),
+        Subject(subject),
+        ActivationId(activationId),
+        Instant.ofEpochMilli(start),
+        Instant.ofEpochMilli(end),
+        response = response,
+        logs = logs,
+        duration = duration,
+        version = SemVer(version),
+        annotations = kindAnnotation ++ causedByAnnotation ++ memoryAnnotation ++ pathAnnotation ++ waitTimeAnnotation ++ initTimeAnnotation,
+        cause = cause.map(value => Some(ActivationId(value))).getOrElse(None))
+    }
+  }
+
+  object ActivationEntry extends DefaultJsonProtocol {
+    implicit val serdes =
+      jsonFormat(
+        ActivationEntry.apply,
+        elasticSearchConfig.schema.name,
+        elasticSearchConfig.schema.subject,
+        elasticSearchConfig.schema.activationId,
+        elasticSearchConfig.schema.version,
+        elasticSearchConfig.schema.start,
+        elasticSearchConfig.schema.end,
+        "result",
+        "statusCode",
+        elasticSearchConfig.schema.duration,
+        elasticSearchConfig.schema.namespace,
+        "kind",
+        "cause",
```

Comment:
All of the hardcoded values for the JSON formatter need to be moved to configuration settings.